spring-core/src/main/java/facade/BookingFacadeImpl.java

```diff
@@ -0,0 +1,149 @@
+package facade;
+
+import model.Event;
+import model.Ticket;
+import model.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import service.EventService;
+import service.TicketService;
+import service.UserService;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+public final class BookingFacadeImpl implements BookingFacade {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BookingFacadeImpl.class);
+
+    private final EventService eventService;
+    private final TicketService ticketService;
+    private final UserService userService;
+
+    public BookingFacadeImpl(EventService eventService, TicketService ticketService, UserService userService) {
+        this.eventService = eventService;
+        this.ticketService = ticketService;
+        this.userService = userService;
+    }
+
+    public Event getEventById(long eventId) {
+        return eventService.retrieveEventById(eventId);
+    }
+
+    public List<Event> getEventsByTitle(String title, int pageSize, int pageNum) {
+        checkPageSize(pageSize, pageNum);
+        final List<Event> events = eventService.retrieveEventByTitle(title);
+        LOG.debug("Events: " + events);
+        return events.isEmpty() ? Collections.emptyList() : listByPage(events, pageSize, pageNum);
+    }
+
+    public List<Event> getEventsForDay(Date day, int pageSize, int pageNum) {
+        checkPageSize(pageSize, pageNum);
+        final List<Event> events = eventService.retrieveEventByDay(day);
+        LOG.debug("Events: " + events);
+        return events.isEmpty() ? Collections.emptyList() : listByPage(events, pageSize, pageNum);
+    }
+
+    public Event createEvent(Event event) {
+        return eventService.addEvent(event);
+    }
+
+    public Event updateEvent(Event event) {
+        return eventService.updateEvent(event);
+    }
+
+    public boolean deleteEvent(long eventId) {
+        return eventService.removeEvent(eventId);
+    }
+
+    public User getUserById(long userId) {
+        return userService.retrieveUserById(userId);
+    }
+
+    public User getUserByEmail(String email) {
+        return userService.retrieveUserByEmail(email);
+    }
+
+    public List<User> getUsersByName(String name, int pageSize, int pageNum) {
+        checkPageSize(pageSize, pageNum);
+        final List<User> users = userService.retrieveUserByName(name);
```

Comment:
final not needed here

Reply:
but why not ? it's a good practice to use final whenever it's possible 
https://stackoverflow.com/questions/137868/using-the-final-modifier-whenever-applicable-in-java

Reply:
this is good practice used for fields with constructor for once creation, but since this is the result of your execution of a method that stores a value, then this is unnecessary all the more it complicates the process of reading the code